crates/debugging/src/trace/collect.rs

```diff
@@ -74,3 +80,28 @@ fn collect_selector(call_trace: &CallTrace, contracts_data: &ContractsData) -> S
         .map(Selector)
         .expect("selector should be present in `ContractsData`")
 }
+
+fn collect_abi(contract_name: &ContractName, contracts_data: &ContractsData) -> Vec<AbiEntry> {
+    let artifacts = contracts_data
+        .get_artifacts(&contract_name.0)
+        .expect("artifact should be present in `ContractsData`");
+
+    let abi = serde_json::from_str::<Value>(&artifacts.sierra)
+        .expect("sierra should be valid json")
+        .get_mut("abi")
+        .expect("abi value should be present in sierra")
+        .take();
+
+    serde_json::from_value(abi).expect("abi value should be valid ABI")
+}
+
+fn collect_calldata(call_trace: &CallTrace, abi: &[AbiEntry]) -> Calldata {
```

Comment:
Bit weird to hide data transformation in `collect_calldata`, maybe rename this?

Reply:
Same remark, maybe `transformed_calldata`?